src/components/MapContainer.tsx

```diff
@@ -25,22 +25,22 @@ function MapContainer({
    */
   const initializeMarkers = useCallback(
     (map: google.maps.Map | null) => {
-      const onClickHandler = (placeId: string) => {
-        getRestaurant(placeId)
-          .then((restaurant: Restaurant) => {
+      const onClickHandler = (restaurantId: string) => {
+        getRestaurant(restaurantId)
+          .then((restaurant: Place) => {
             restaurantDispatch({
               type: RestaurantDispatch.TOGGLE_CARD,
-              payload: { placeId, restaurant },
+              payload: { restaurantId, restaurant },
             });
           })
           .catch(() => restaurantDispatch({ type: RestaurantDispatch.HIDE_CARD }));
       };
 
       if (map) {
-        promotionsState.data.forEach(({ lat, lon, placeId }) => {
-          const position = { lat, lng: lon };
+        promotionsState.data.forEach(({ restaurant }) => {
+          const position = { lat: restaurant.lat, lng: restaurant.lon };
           const marker = new google.maps.Marker({ position, map });
-          marker.addListener('click', () => onClickHandler(placeId));
+          marker.addListener('click', () => onClickHandler(restaurant.id));
         });
```

Comment:
I was thinking rather than using `placeId`, we use the `id` of the restaurant. This is because each restaurant is associated with a `placeId`.

Reply:
Lol my random latitude and longitude in the BE has made the restaurants spread out throughout the world
![image](https://user-images.githubusercontent.com/49849754/110232455-8d130d00-7ed2-11eb-97de-32efa509bf86.png)